src/zap/zapimage.cpp

```diff
@@ -681,8 +681,17 @@ void ZapImage::SetVersionInfo(CORCOMPILE_VERSION_INFO * pVersionInfo)
 
 void ZapImage::SetDependencies(CORCOMPILE_DEPENDENCY *pDependencies, DWORD cDependencies)
 {
-    m_pDependencies = new (GetHeap()) ZapDependencies(pDependencies, cDependencies);
-    m_pHeaderSection->Place(m_pDependencies);
+    if (IsReadyToRunCompilation())
+    {
+        ZapNode* pBlob = new (GetHeap()) ZapDependencies(pDependencies, cDependencies);
```

Comment:
Here I need to figure out how to filter the `CORCOMPILE_DEPENDENCY*` into things that are part of the version bubble.

Hence this pull request is a draft.

Reply:
Mm, seems like this is not needed. Everything is part of the current version bubble for crossgen.

https://github.com/dotnet/coreclr/blob/8974a699899bdc2cc5687504e1ada606ac803e9b/src/vm/ceeload.cpp#L3395-L3396